src/utils/warnings.ts

```diff
@@ -0,0 +1,17 @@
+/*!
+ * Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved.
+ */
+
+let warnings: string[][] = [];
+
+export function recordWarning(...warn: string[]) {
+  warnings.push(warn);
+}
+
+export function getWarnings(): string[] | undefined {
+  if (warnings.length > 0) {
+    const result = ['⚠️  There were the following warnings:', ...warnings.map(warn => `- ${warn.join('\n  ')}`)];
```

Comment:
'There were the following warnings:' sounds a little german. What do you think about just 'Warnings:'?